### DISCUSSION TITLE
Pledge to keep supporting dat:// as read-only

### DISCUSSION THREAD

**Discussion by despens** [beakerbrowser/beaker#1604] [General] [answered]
Dear Beaker people,

It would be pretty sad to loose sites currently published on dat://

If you're not at least supporting a read-only display of dat:// resources, you're basically asking everyone who ever published on dat:// to point people to a legacy version of Beaker to access their material or to completely redo their sites and links. This will be especially difficult for collaborative projects spread across several dat:// datasets, for instance *Floating Swarm* by Rafäel Bastide. dat://de037e790893b7113b93bc523c4cf7671acdbc370bfa0acc00e0b3781a83b56b/

There also doesn't seem to be a DNS→Hyperdrive or .well-known→Hyperdrive plan in place yet that for instance was used on https://digitalfolklore.org that points to a dat:// site with different resources available at dat://f1a5ed8cd08dc9704d6261c899bbd0f5a0851e596f132d6edc562c1a4948c43e/

I have also figured out that on my System (Ubuntu Linux using AppImage versions of Beaker) it is not apparent how to run legacy versions of Beaker if the latest version is installed. All my 0.8.x AppImages now launch the 1.0 pre-release.

Don't become the Yahoo of p2p! :wink:

**Comment by raphaelbastide** [comment]
I agree with @despens. Here is a non-exhaustive list of dat-related projects that just can’t be lost or broken, most of them can’t be changed or adapted without changing their nature:

- https://davidbenque.com/projects/almanac/ a work in progress by @davidbenque
- https://peer-to.peer-to-peer-web.com/ an online exhibition curated by @jondashkyle containing 16 online pieces, most of them are based on dat
- http://olia.lialina.work/ a major piece of net art by Olia Lialina, that depends on dat
- Vogelhaus, a dat server with art projects by HFG students ([screenshot](https://d2w9rnfcy7mm78.cloudfront.net/3804391/original_c99cfb860b6eb4703bcfba1a0ab3d77c.png)) dat://d0cdd1916150c0e9a6b6b1c66df16a2c5808c1135952d7bed925c4146eff7744/
- Renaming the web, an art installation depending on Beaker and dat https://raphaelbastide.com/renaming-the-web/ dat://36652fc7932d5c8f59cb03c0d9325a7a123e560268815269ba747e477b526f41
- All the projects I missed or don’t know

Thank you @despens for pointing this issue. Even though Beaker+Hyperdrive sounds exciting, I think Beaker should not deny its past, on the contrary, it is very much in the P2P spirit to make sure [content remain accessible](https://twitter.com/pfrazee/status/1002827413900988418) in hyperspace and hypertime :)

**Comment by pfrazee** [comment]
We are listening to yall here and trying to make the best call we can. It's a time juggling act. We knew losing the existing dat:// content sucks but we're also trying to keep a forward pace going -- after taking over a year to just get out the 1.0 beta, we didn't want to delay longer for legacy dat support.

We want people to migrate to Hypercore and our main goal with legacy support is to make that possible. (It would help if I didn't keep shipping a broken dat converter on windows.) 

That said, I know migrating takes some time and it would be helpful to be able to still view dats. I *think* I can hack together a way for the dat converter to access the network (instead of only locally-saved dats as right now) and also support dat-dns. I also think I can get the web pages to display rather than showing only the converter interface. The converter could move to some other browser UI. What that doesn't include is the old `DatArchive` Web APIs, so some of the dat pages won't work correctly.

Do yall feel like that would be enough?

**Reply by martinheidegger** [reply]
We have dat urls on business cards with dns domains and advice to use BeakerBrowser to browse it. In lieu of redirect mechanism from dat:// to hyper:// we can throw these away.

**Comment by jondashkyle** [comment]
Gonna chime in real quick here.

Of course, archival was always a critical part of my interest in `dat://` and Beaker. So yeah — I’d love to see some form of continued support. But I also realize that Beaker is somewhat of a personal gesture. I don’t say this reductionistically. There’s something pure about it. So; I don’t expect anything from Paul as a creator and maintainer, or any of the other contributors. It’s all experimental in nature. I think if someone is willing to contribute time to implementing read-only support, awesome! You’re now investing similar time and resources as Paul would making that a reality. I also understand Paul’s ambition to move forward.

Has been a lot of fun watching this mature, and congrats on the release!

**Comment by pfrazee** [comment, marked as answer]
Good news - master branch now has read-only browsing for dat:// sites. We should release it this week (Thursday if we maintain our regular schedule). 

![ezgif-4-5287274a272e](https://user-images.githubusercontent.com/1270099/83580828-aaa9e500-a502-11ea-9f3a-50c8897f0134.gif)

The updates will also make it much easier to test the dat:// converter, so hopefully fewer releases will have breaks in the code.

**Reply by despens** [reply]
That's my boy Paul right there

**Reply by raphaelbastide** [reply]
That looks great Paul!

**Reply by davidbenque** [reply]
Thank you for listening Paul!

**Reply by pfrazee** [reply]
<3 yall

**Reply by raphaelbastide** [reply]
Just tested it with Beaker 1.0.0-prerelease.4, it’s wonderful to see dat archives in Beaker 1!

**Reply by jondashkyle** [reply]
Awesome to see, nice going Paul!
